src/Functions/likePatternToRegexp.h

```diff
@@ -21,52 +21,84 @@ inline String likePatternToRegexp(std::string_view pattern)
     const char * const end = pattern.begin() + pattern.size();
 
     if (pos < end && *pos == '%')
-        ++pos;
+    {
+        while (++pos < end && *pos == '%')
+        {
+        }
+    }
     else
+    {
         res = "^";
-
-    while (pos < end)
+    }
+    while (pos != end)
     {
         switch (*pos)
         {
-            case '^': case '$': case '.': case '[': case '|': case '(': case ')': case '?': case '*': case '+': case '{':
+            case '^':
+            case '$':
+            case '.':
+            case '[':
+            case '|':
+            case '(':
+            case ')':
+            case '?':
+            case '*':
+            case '+':
+            case '{':
                 res += '\\';
                 res += *pos;
+                ++pos;
                 break;
             case '%':
-                if (pos + 1 != end)
+                ++pos;
+                if (pos != end)
+                {
                     res += ".*";
+                }
                 else
+                {
                     return res;
+                }
                 break;
             case '_':
+                ++pos;
```

Comment:
For consistency accross the `case`s, could `++pos` be deferred as much as possible? E.g. here, it could be done immediately before `break`.

Reply:
@rschu1ze done.